layer/frame_time_layer.cc

```diff
@@ -103,33 +136,18 @@ VKAPI_ATTR VkResult FrameTimeLayer_QueuePresentKHR(
   uint64_t frames_elapsed = layer_data->IncrementFrameNum();
   uint64_t exit_frame_num = layer_data->GetExitFrameNum();
   // If the layer should make Vulkan application exit after this frame.
-  if (exit_frame_num != FrameTimeLayerData::kInvalidFrameNum &&
-      frames_elapsed >= exit_frame_num) {
-    if (const char* finish_indicator_file = getenv(kFinishFileEnvVar)) {
-      // Create the application finish indicator file and write the current time
-      // there. This is to aid debugging when the modification time can be lost
-      // when sending the file over a wire.
-      if (FILE* finish_file = fopen(finish_indicator_file, "w")) {
-        std::time_t curr_time = std::time(nullptr);
-        std::tm tm = *std::localtime(&curr_time);
-        std::ostringstream oss;
-        oss << std::put_time(&tm, "%c %Z");
-        fprintf(finish_file, "Stadia Frame Time Layer\nTerminated on %s\n",
-                oss.str().c_str());
-        fclose(finish_file);
-      }
-    }
-
+  if (frames_elapsed == exit_frame_num) {
     fprintf(
         stderr,
         "Stadia Frame Time Layer: Terminating application after frame %" PRIu64
         "\n",
         frames_elapsed);
     fflush(stderr);
+
     // _Exit will bring down the parent Vulkan application without running any
     // cleanup. Resources will be reclaimed by the operating system.
-    // Delete to force the destructor to save the layer log files.
-    delete layer_data;
+    // Call the destructor manually to save the layer log files.
+    layer_data->~FrameTimeLayerData();
```

Comment:
I'd prefer to see this call layer_data->CreateFinishIndicatorFile("FRAME_TIME_LAYER_TERMINATED") rather than manually calling the destructor. Then the destructor can also just be  CreateFinishIndicatorFile("APPLICATION_EXIT);

Reply:
Calling the destructor ensures that the log file is closed. But since every print is now followed by a flush, I guess this should be fine?

Reply:
Correct, it's fine to _exit without closing the file because of the fflush() calls. Alternatively, consider making the static local variables unique_ptr's that you can reset here (that way you can manually trigger deletion here and when the layer is unloaded). Technically calling the destructor as in your original implementation is fine, but it's just surprising to see a direct destructor call for something that isn't created with placement new